docs(parameters): snippets split, improved, and lint

leandrodamascena opened the pull request:
<!-- markdownlint-disable MD041 MD043 -->
**Issue number:** #1368 

## Summary

### Changes

Changes snippet fictitious names to real code snippet filenames for easier discovery and collaboration to improve/correct document snippets.

Some examples have also been refactored so that the codes are complete and there are no lint/pre-commit hooks issues.

### User experience

Before:
![image](https://user-images.githubusercontent.com/4295173/193474191-c80214f7-f519-477d-a8ab-8c57c2f31723.png)

After:
![image](https://user-images.githubusercontent.com/4295173/193474202-6af44a6e-c020-4e5d-bbf1-e0cff1e5242b.png)

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.


-----
[View rendered docs/utilities/parameters.md](https://github.com/leandrodamascena/aws-lambda-powertools-python/blob/chore/parameters/docs/utilities/parameters.md)

leandrodamascena committed:
chore(parameters): refactoring examples

leandrodamascena committed:
chore(parameters): adding cache examples

leandrodamascena committed:
chore(parameters): adding builtin examples and fix mypy errors

leandrodamascena committed:
chore(parameters): adding all providers with custom config

leandrodamascena committed:
chore(parameters): adding custom provider

leandrodamascena committed:
chore(parameters): adding custom provider

leandrodamascena committed:
chore(parameters): adding custom provider

leandrodamascena committed:
chore(parameters): adding DynamoDB provider

leandrodamascena committed:
chore(parameters): adding tests - mock

leandrodamascena committed:
chore(parameters): adding more tests

leandrodamascena committed:
chore(parameters): adding appconfig builtin provider

leandrodamascena committed:
chore(parameters): adding final examples

leandrodamascena committed:
chore(parameters): adding final examples

leandrodamascena committed:
chore(parameters): highlights - small changes

leandrodamascena committed:
Merge branch 'develop' into chore/parameters

heitorlessa committed:
fix(docs): broken link in boto3_session

heitorlessa committed:
fix(docs): improve constrast ratio

heitorlessa committed:
fix(docs): typing and correctness on null endpoint

leandrodamascena committed:
Merge branch 'develop' into chore/parameters

Signed-off-by: Leandro Damascena <leandro.damascena@gmail.com>

leandrodamascena committed:
(merge): merging

rubenfonseca commented on docs/utilities/parameters.md:
```diff
@@ -37,7 +40,7 @@ This utility requires additional permissions to work as expected.
 ### Fetching parameters
 
-You can retrieve a single parameter  using `get_parameter` high-level function.
+You can retrieve a single parameter  using the `get_parameter` high-level function.
```
```suggestion
You can retrieve a single parameter using the `get_parameter` high-level function.
```

rubenfonseca commented on docs/utilities/parameters.md:
```diff
@@ -123,93 +126,95 @@ def handler(event, context):
 
 You can fetch secrets stored in Secrets Manager using `get_secrets`.
```
```suggestion
You can fetch secrets stored in Secrets Manager using `get_secret`.
```

leandrodamascena replied:
Done

rubenfonseca commented on docs/utilities/parameters.md:
````diff
@@ -334,94 +313,50 @@ DynamoDB provider can be customized at initialization to match your table struct
-	value_attr="MyvalueAttr"
-)
+=== "builtin_provider_dynamodb_custom_fields.py"
+    ```python hl_lines="3 5 9 10 15"
````
On the rendered docs the highlight seems wrong. Can you please review it?

leandrodamascena replied:
Yep! Fixed.

rubenfonseca reviewed (changes requested):
This is amazing work @leandrodamascena thank you! I left some small comments but I ask you the following two things:

- can you please render the docs locally and manually check all the highlights to see if they make sense?
- can you rebase the branch with `develop`?

We should be able to merge this very soon

leandrodamascena committed:
(merge): merging

leandrodamascena committed:
(chore): addressing Ruben's feedbacks

leandrodamascena committed:
(chore): import errors

leandrodamascena committed:
(merge): merging

leandrodamascena commented:
> This is amazing work @leandrodamascena thank you! I left some small comments but I ask you the following two things:
> 
> * can you please render the docs locally and manually check all the highlights to see if they make sense?
> * can you rebase the branch with `develop`?
> 
> We should be able to merge this very soon

Hei Ruben! Thanks a lot for the feedback. I addressed all issues you mentioned and I moved 2 code snippet from `parameters.md` file to example folder.

leandrodamascena committed:
(merge): merging

rubenfonseca reviewed (approved):
Great work!
